server/bleep/src/background.rs

```diff
@@ -213,17 +223,36 @@ impl IndexWriter {
             }
         };
 
-        creds.sync(app.clone(), repo).await
+        let synced = creds.sync(app.clone(), repo.clone()).await;
+        if let Err(RemoteError::RemoteNotFound) = synced {
+            self.0
+                .repo_pool
+                .get_mut(&repo)
+                .unwrap()
+                .value_mut()
+                .sync_status = SyncStatus::RemoteRemoved;
+
+            error!(?repo, "remote repository removed; disabling local syncing");
+
+            // we want indexing to pick this up later and handle the new state
+            // all local cleanups are done, so everything should be consistent
+            return Ok(());
+        }
+
+        Ok(synced?)
     }
 
-    fn delete_repo(
+    async fn delete_repo_indexes(
         &self,
         repo: &Repository,
         writers: &indexes::GlobalWriteHandleRef<'_>,
     ) -> anyhow::Result<()> {
         let IndexWriter(Application { config, .. }) = self;
 
         repo.delete_file_cache(&config.index_dir)?;
+        if !matches!(repo.remote, RepoRemote::None) {
+            tokio::fs::remove_dir_all(&repo.disk_path).await?;
+        }
```

Comment:
Perhaps add a comment that this does the actual cleanup of deleted repos. Also are we sure we want to delete the local workspace? Should we put this behind an option?

Reply:
This doesn't delete any local repos, only repos managed by bloop itself. It also only cleans up the indexes/state when the user explicitly flips it into `Removed`, otherwise we'll keep these around.

Not sure what you mean by putting it behind an option?